lib/utils/ObjectUtils.ts

```diff
@@ -26,4 +26,24 @@ export class ObjectUtils {
     const fieldValues = data.map((item) => item[fieldName]);
     return Array.from(new Set(fieldValues));
   }
+  /**
+   * Receives an object and clone deep, return the cloned object
+   */
+  public static cloneDeep<T>(obj: T): T {
+    if (obj === null || typeof obj !== 'object') {
+      return obj;
+    }
+
+    if (Array.isArray(obj)) {
```

Comment:
This expression will never evaluate because of the expression above `!== object`.

It looks like this is always used on payload. Any reason to not simply do a JSON.parse/stringify?

Reply:
I'll change that way :)
I thought you might not want it that way.